### ISSUE TITLE
Compiling `diesel-migrations` for FreeBSD fails.

### ISSUE THREAD

**Issue by gahag** [cross-rs/cross#520]
Compiling `diesel-migrations` with `cross build --target=x86_64-unknown-freebsd` results in the following error:

```   Compiling diesel_migrations v1.4.0
error: libthr.so.3: cannot open shared object file: No such file or directory
  --> /cargo/registry/src/github.com-1ecc6299db9ec823/diesel_migrations-1.4.0/src/lib.rs:82:1
   |
82 | extern crate migrations_macros;
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

error: could not compile `diesel_migrations`
```

`Cargo.toml`:
```[package]
name = "ohgodwhy"
version = "0.1.0"
authors = ["gahag <gabriel.s.b@live.com>"]
edition = "2018"

[dependencies]
diesel = { version = "1.4", features = ["sqlite"] }
diesel_migrations = { version = "1.4", features = ["sqlite"] }
```
`main.rs`:
```
fn main() {
    println!("Hello, world!");
}
```

Is this an issue with the cross' FreeBSD setup, or with diesel-migrations?

**Comment by gahag** [comment]
For some odd reason, I have two projects that use `diesel_migrations`. One of them runs fine, while the other fails with the same error as the supplied minimal example. I have no idea why one of them works.

**Comment by tesaguri** [comment]
This seems to be caused by `SQLITE3_LIB_DIR` environment variable in the FreeBSD container, which, I guess, causes the procedural macro to attempt to link against the target libsqlite3, instead of the host's one.

I have filed a fix for this issue: #549.

**Comment by gahag** [comment]
It's great that you found the issue. I'm excited to try on this patch.
